Removed some dependencies on com.sun

lutovich opened the pull request:
This PR includes 4 commits:
- Rewritten/removed tests that use JDI and breakpoints
- Removed infrastructure for JDI tests
- Removed unfinished & unused Neo4j JConsole plugin
- Removed `RetryOnGcRule`

lutovich committed:
Rewritten/removed tests that use JDI and breakpoints

 * removed TestConcurrentRotation. it was supposed to verify that it is
   possible to load legacy index while rotating lucene transaction log.
   Currently we only have a single transaction log and rotation happens
   in a dedicated checkpointer thread concurrently with transactions.
   That is why this test does not seem to be relevant any more

 * TestPropertyReadOnNewEntityBeforeLockRelease rewritten in stochastic
   way to not use JDI and breakpoints

 * Removed breakpoints from IdGeneratorRebuildFailureEmulationTest.
   Breakpoint enabling annotations were just leftovers and not actually
   used.

lutovich committed:
Removed infrastructure for JDI tests

Initial motivation for doing this change was that JDI test classes
depended on proprietary com.sun packages which are not available on IBM JDK.

At the same time there were only two tests that used this infrastructure.
One of them was obsolete and other one was simply rewritten in a
stochastic way.

lutovich committed:
Removed unfinished Neo4j JConsole plugin

Plugin did not seem to be used anywhere and was not entirely
implemented - contained many old TODO items. It also depended on com.sun
classes that are not present in IBM JDK.

chrisvest commented on community/kernel/src/test/java/org/neo4j/kernel/impl/store/IdGeneratorRebuildFailureEmulationTest.java:
```diff
@@ -75,7 +73,6 @@ protected void emulateFailureOnRebuildOf( NeoStores neoStores )
     }
 
-    @BreakpointTrigger
     private void performTest() throws Exception
```
Do these tests still work like before? I'm not seeing what the break points were actually doing in this case.

A small additional cleanup could be to pass the id file name as a parameter, instead of computing it from the stack trace.

lutovich replied:
Yes, they still work. Breakpoint annotations were not used.

lutovich committed:
Removed RetryOnGcRule

It depended on proprietary com.sun classes and did not work on IBM JDK.
Having rule that retries clustering tests if GC happened does not seem to be
useful because it just masks real clustering issues.

chrisvest commented:
Praise the sun!

davidegrohmann commented:
![Clean all the things!](https://sunaycleaning.files.wordpress.com/2012/05/bottom_border_clean_shorter1.jpg)
